src/components/Collection/Tags/TagDetailsModal.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { mdiOpenInNew } from '@mdi/js';
+import Icon from '@mdi/react';
+import cx from 'classnames';
+
+import AnidbDescription from '@/components/Collection/AnidbDescription';
+import ModalPanel from '@/components/Panels/ModalPanel';
+import { useFilteredSeriesInfiniteQuery } from '@/core/react-query/filter/queries';
+import useFlattenListResult from '@/hooks/useFlattenListResult';
+
+import type { TagType } from '@/core/types/api/tags';
+
+const TagDetailsModal = React.memo(({ onClose, show, tag }: { show: boolean, tag?: TagType, onClose: () => void }) => {
+  const { data: seriesDataList, fetchNextPage, isFetchingNextPage, isSuccess } = useFilteredSeriesInfiniteQuery(
+    {
+      pageSize: 50,
+      filterCriteria: {
+        ApplyAtSeriesLevel: true,
+        Expression: {
+          Parameter: tag?.Name ?? '',
+          Type: tag?.Source === 'User' ? 'HasCustomTag' : 'HasTag',
+        },
+      },
+    },
+    show,
+  );
+  const [seriesData, seriesCount] = useFlattenListResult(seriesDataList);
+
+  if (!isFetchingNextPage && seriesData.length !== seriesCount) {
+    fetchNextPage().catch(() => {});
+  }
```

Comment:
could just make it pageSize: 0 if we are loading everything anyway

Reply:
Just to provide justificication for my original approach - I wasn't too sure what the tradeoff might be between doing a potentially large/"slow" network request for users with larger collections vs many smaller repeated requests. I decided to err on the side of caution so went to implement this using a page size of 1. Since everything seemed to remain responsive, I kept it as is, albeit with a more reasonable default page size.

If there's no need for performance concerns on this front then happy days, should make things simpler for sure.

Reply:
In that case I would argue that the rows should also be virtualized. But again, in a different PR. Leave this as is